crates/lib/src/compiler/quilc.rs

```diff
@@ -33,12 +33,19 @@ pub const DEFAULT_COMPILER_TIMEOUT: f64 = 30.0;
 /// recoverable at runtime. This function can fail generally if the provided ISA cannot be converted
 /// into a form that `quilc` recognizes, if `quilc` cannot be contacted, or if the program cannot
 /// be converted by `quilc`.
+#[cfg_attr(
+    feature = "tracing",
+    tracing::instrument(skip(client), level = "trace")
+)]
 pub fn compile_program(
     quil: &str,
     isa: TargetDevice,
     client: &Qcs,
     options: CompilerOpts,
 ) -> Result<Program, Error> {
+    #[cfg(feature = "tracing")]
+    tracing::debug!(%quil, compiler_options=?options, "compiling quil program with quilc",);
```

Comment:
These programs can be massive - tens of thousands of lines long. Probably best not to log

Reply:
Good point.